sonar-python-plugin/src/main/java/org/sonar/plugins/python/cpd/PythonCpdAnalyzer.java

```diff
@@ -42,25 +43,24 @@ public void pushCpdTokens(InputFile inputFile, PythonVisitorContext visitorConte
     AstNode root = visitorContext.rootTree();
     if (root != null) {
       NewCpdTokens cpdTokens = context.newCpdTokens().onFile(inputFile);
-      for (Token token : root.getTokens()) {
-        if (!isIgnoredType(token.getType())) {
+      List<Token> tokens = root.getTokens();
+      for (int i = 0; i < tokens.size(); i++) {
+        Token token = tokens.get(i);
+        TokenType nextTokenType = i + 1 < tokens.size() ? tokens.get(i + 1).getType() : GenericTokenType.EOF;
+        if (!isIgnoredType(token.getType(), nextTokenType)) {
           TokenLocation location = new TokenLocation(token);
-          cpdTokens.addToken(location.startLine(), location.startLineOffset(), location.endLine(), location.endLineOffset(), getImage(token));
+          cpdTokens.addToken(location.startLine(), location.startLineOffset(), location.endLine(), location.endLineOffset(), token.getValue());
         }
       }
       cpdTokens.save();
     }
   }
 
-  private boolean isIgnoredType(TokenType type) {
-    return type.equals(PythonTokenType.NEWLINE) ||
-      type.equals(PythonTokenType.DEDENT) || // TODO equals { } ???
+  private static boolean isIgnoredType(TokenType type, TokenType nextTokenType) {
+    return (type.equals(PythonTokenType.NEWLINE) && !nextTokenType.equals(PythonTokenType.DEDENT)) ||
```

Comment:
I would extract this condition into a dedicated method, maybe called `isNotNewLineWithIndentationChange()` ? maybe it could even be completely out of the `isIgnoredType` method and simply called from [line #50](https://github.com/SonarSource/sonar-python/pull/181/files#diff-a3ead01d78a47faaa0dd1cfeb5365f04R50).